vue/components/organisms/configurator/configurator.vue

```diff
@@ -486,6 +360,177 @@ export const Configurator = {
+                if (this.ignoreBus) return;
+
+                // avoid infinite loop, by checking if the frame
+                // is the one we're currently on
+                if (this.frameData === frame) return;
+
+                // in case the configurator is not currently ready
+                // then avoids the operation (returns control flow)
+                if (!this.configurator || !this.configurator.ready) {
+                    return;
+                }
+
+                try {
+                    // triggers the async change frame operation on
+                    // the current configurator
+                    await this.configurator.changeFrame(frame, {
+                        type: null,
+                        duration: null
+                    });
+                } catch (error) {
+                    // calls the registered callback handler for the
+                    // error (default implementation is a simple re-throw)
+                    this.onError(error);
+                }
+            });
+
+            this.$bus.bind("show_frame", async (frame, options) => {
+                // in case the global bus should be ignore nothing is
+                // done as a consequence of a changed frame
+                if (this.ignoreBus) return;
+
+                // avoid infinite loop, by checking if the frame
+                // is the one we're currently on
+                if (this.frameData === frame) return;
+
+                // in case the configurator is not currently ready
+                // then avoids the operation (returns control flow)
+                if (!this.configurator || !this.configurator.ready) {
+                    return;
+                }
+
+                try {
+                    // triggers the async change frame operation on
+                    // the current configurator
+                    await this.configurator.changeFrame(frame, options);
+                    this.frameData = frame;
+                } catch (error) {
+                    // calls the registered callback handler for the
+                    // error (default implementation is a simple re-throw)
+                    this.onError(error);
+                }
+            });
+
+            this.$bus.bind("highlight_part", part => {
+                if (this.ignoreBus) return;
+                if (this.configurator && this.configurator.ready) {
+                    this.configurator.highlight(part);
+                }
+            });
+
+            this.$bus.bind("lowlight_part", part => {
+                if (this.ignoreBus) return;
+                if (this.configurator && this.configurator.ready) {
+                    this.configurator.lowlight(part);
+                }
+            });
+
+            this.resize(this.size, this.width, this.height);
+
+            if (this.configuratorType === "csr") {
+                this.ripeInstance.trigger("post_config", this.ripeInstance.loadedConfig);
+            }

```

Comment:
Why is this needed? It's because of white not using ripe-sdk correctly right?

Reply:
This is because in ripe-white implementation `post_config` is triggered before the `_registerConfigHandlers` method in CSR which CSR uses to initialize the initials and render.

This has to do with the fact that we're binding a CSR configurator after the `post_config` already happened so we're triggering a `post_config` manually.

![image](https://user-images.githubusercontent.com/24736423/221549797-b1344a55-11c8-4d38-bb60-e220bfc2bb1f.png)

I improved the guard and added a comment to make it clear but feel free to share if you have any suggestion on improving this